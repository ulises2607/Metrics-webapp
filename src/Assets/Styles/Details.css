.details-title {
  color: #fce4ef;
  font-size: 1rem;
  font-weight: 600;
  background-color: #df4783;
  display: flex;
  align-items: center;
  padding-left: 0.5rem;
  height: 2rem;
}

.detail-container {
  background-color: #ec4c8a;
}
