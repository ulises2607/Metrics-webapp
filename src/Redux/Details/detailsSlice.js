import { createSlice } from "@reduxjs/toolkit";
import { ejemplo } from "../../database/database";
import { createAsyncThunk } from "@reduxjs/toolkit";
const API_KEY = '3e341f1094147b1dd3b25d397b2c892e';


const initialState = {
    cities: ejemplo,
};

const getData = async (longitude, latitude) => {
    try {
        const response = await fetch(`https://api.openweathermap.org/data/2.5/air_pollution/forecast?lat=${latitude}&lon=${longitude}&appid=${API_KEY}`);
        const data = await response.json();
        return data.list[0].components;
    } catch (error) {
        console.log(error, 'error en el getData');
    }
}

export const fetchDetail = createAsyncThunk('home/fetchHome', async ( country ) => {
    try {
        const response = await fetch(`https://api.openweathermap.org/geo/1.0/direct?q=${country}&appid=${API_KEY}`);
        const data = await response.json();
        console.log(data, 'data');
        const { lat, lon } = data[0];
        const data2 = await getData(lon, lat);
        console.log(data2, 'data2');
        return data2;
    } catch (error) {
        console.log(error, 'error en el fetchhome');
    }

    return final;
});

const detailsSlice = createSlice({
    name: "detail",
    initialState,
    reducers: {

    },
    extraReducers: {
        [fetchDetail.fulfilled]: (state, action) => {
            state.details = action.payload;
        }
    }
});


export default detailsSlice;

