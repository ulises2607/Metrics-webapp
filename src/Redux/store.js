import { configureStore } from "@reduxjs/toolkit";
import homeSlice from "./Home/HomeSlice";

const store = configureStore({
    reducer: {
        home: homeSlice.reducer,
    }
});

export default store;