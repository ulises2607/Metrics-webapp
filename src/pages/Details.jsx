import React, { useEffect } from 'react';
import { useParams } from 'react-router';
import { useDispatch, useSelector } from 'react-redux';
import { fetchDetail } from '../Redux/Details/detailsSlice';
import DetailsCard from '../Components/DetailsCard';

const Details = () => {
  const { country } = useParams();
  const dispatch = useDispatch();

  const cities = useSelector(state => state.details.cities);

  return (
    <div className='details-container'>
      <h2>{country}</h2>
      <ul>
        {cities.map(countryData => {
          if (countryData.nombre === country) {
            return countryData.provinces.map(province => (
              <li key={province.nombre}>
                <DetailsCard name={province.nombre} 
                imp_city={province.important_city}
                show = {province.show} 
                details={province.details}
                />
              </li>
            ));
          }
          return null;
        })}
      </ul>
    </div>
  );
}

export default Details;
