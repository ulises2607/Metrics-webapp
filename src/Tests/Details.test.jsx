import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Details from '../pages/Details';
import store from '../Redux/store';

test('Snapshot del componente Details', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Details />
      </MemoryRouter>
    </Provider>,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
