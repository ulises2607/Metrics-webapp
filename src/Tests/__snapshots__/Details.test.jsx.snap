// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot del componente Details 1`] = `
<div
  className="details-container"
>
  <div
    className="header-container"
  >
    <div
      className="img-header-container"
    >
      <img
        alt="logo"
        className="logo-header"
        src="pngegg.png"
      />
    </div>
    <div
      className="title-header-container"
    >
      <h1
        className="title-header"
      >
        South America air polution
      </h1>
    </div>
  </div>
  <div
    className="details-title"
  >
    <h2>
      Provinces
    </h2>
  </div>
  <div
    className="search-container"
  >
    <input
      onChange={[Function]}
      placeholder="Buscar provincias..."
      type="text"
      value=""
    />
  </div>
  <ul
    className="province-list"
  />
</div>
`;
